src/include/sof/lib/notifier.h

```diff
@@ -19,42 +19,31 @@ struct sof;
 #define NOTIFIER_TARGET_CORE_ALL_MASK	0xFFFFFFFF
 
 enum notify_id {
-	NOTIFIER_ID_CPU_FREQ = 0,
-	NOTIFIER_ID_SSP_FREQ,
-	NOTIFIER_ID_KPB_CLIENT_EVT,
-	NOTIFIER_ID_DMA_DOMAIN_CHANGE,
+	NOTIFIER_ID_CPU_FREQ = 0,		/* struct clock_notify_data * */
+	NOTIFIER_ID_SSP_FREQ,			/* struct clock_notify_data * */
+	NOTIFIER_ID_KPB_CLIENT_EVT,		/* struct kpb_event_data * */
+	NOTIFIER_ID_DMA_DOMAIN_CHANGE,		/* struct dma_chan_data * */
+	NOTIFIER_ID_MAX
```

Comment:
It looks like `ID_MAX` is the first invalid ID, so, it isn't the maximum (valid) ID any more, maybe call it `ID_NUMBER` or `ID_COUNT`? That's also how you use that value, it seems.

Reply:
done